trajopt_optimizers/trajopt_sqp/test/cart_position_optimization_unit.cpp

```diff
@@ -0,0 +1,181 @@
+    }
+
+    mod_url = package_path + mod_url;
+  }
+
+  return mod_url;
+}
+
+class CartPositionOptimization : public testing::TestWithParam<const char*>
+{
+public:
+  // 1) Create the problem
+  ifopt::Problem nlp_;
+
+  Eigen::VectorXd joint_target_;
+
+  void SetUp() override
+  {
+    if (DEBUG)
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_DEBUG);
+    else
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_NONE);
+
+    // 1)  Load Robot
+    boost::filesystem::path urdf_file(std::string(TRAJOPT_DIR) + "/test/data/arm_around_table.urdf");
+    boost::filesystem::path srdf_file(std::string(TRAJOPT_DIR) + "/test/data/pr2.srdf");
+    tesseract_scene_graph::ResourceLocator::Ptr locator =
+        std::make_shared<tesseract_scene_graph::SimpleResourceLocator>(locateResource);
+    auto tesseract = std::make_shared<tesseract::Tesseract>();
+    tesseract->init(urdf_file, srdf_file, locator);
+
+    // Extract necessary kinematic information
+    auto forward_kinematics = tesseract->getFwdKinematicsManager()->getFwdKinematicSolver("right_arm");
+    tesseract_environment::AdjacencyMap::Ptr adjacency_map = std::make_shared<tesseract_environment::AdjacencyMap>(
+        tesseract->getEnvironment()->getSceneGraph(),
+        forward_kinematics->getActiveLinkNames(),
+        tesseract->getEnvironment()->getCurrentState()->link_transforms);
+    auto kinematic_info = std::make_shared<trajopt::CartPosKinematicInfo>(
+        forward_kinematics, adjacency_map, Eigen::Isometry3d::Identity(), forward_kinematics->getTipLinkName());
+
+    // Get target position
+    Eigen::VectorXd start_pos(forward_kinematics->numJoints());
+    start_pos << 0.0, 0, 0, -1.0, 0, -1, -0.00;
+    if (DEBUG)
+      std::cout << "Joint Limits:\n" << forward_kinematics->getLimits().transpose() << std::endl;
+
+    auto target_pose = Eigen::Isometry3d::Identity();
+    joint_target_ = start_pos;
+    forward_kinematics->calcFwdKin(target_pose, joint_target_);
+    //    auto world_to_base =
+    //        tesseract->getEnvironment()->getCurrentState()->link_transforms.at(forward_kinematics->getBaseLinkName());
+    //    target_pose = world_to_base * target_pose;
+
+    // 3) Add Variables
+    std::vector<trajopt::JointPosition::Ptr> vars;
+    for (int ind = 0; ind < 1; ind++)
+    {
+      auto zero = Eigen::VectorXd::Zero(7);
+      auto var = std::make_shared<trajopt::JointPosition>(
+          zero, forward_kinematics->getJointNames(), "Joint_Position_" + std::to_string(ind));
+      vars.push_back(var);
+      nlp_.AddVariableSet(var);
+    }
+
+    // 4) Add constraints
+    for (const auto& var : vars)
+    {
+      auto cnt = std::make_shared<trajopt::CartPosConstraint>(target_pose, kinematic_info, var);
+      nlp_.AddConstraintSet(cnt);
+      //    cnt->LinkWithVariables(nlp_.GetOptVariables());
+      //    auto cost = std::make_shared<trajopt::SquaredCost>(cnt);
+      //    nlp_.AddCostSet(cost);
```

Comment:
Should the commented code be removed or have a TODO?

Reply:
That was there from when I was playing with adding it as a cost as well. I guess I can remove it

Reply:
You could make the test take a bool where you run both cases.